api/jobs.go

```diff
@@ -1544,3 +1544,10 @@ func (j *Jobs) ActionExec(ctx context.Context,
 
 	return s.run(ctx)
 }
+
+type JobsStatusesRequest struct {
+	Jobs []struct { // TODO: proper type
+		ID        string `json:"id"`
+		Namespace string `json:"namespace"`
+	} `json:"jobs"` // TODO: unkeyed?
+}
```

Comment:
We have the type here but not a function that lets us use it in the Go API. If we had that, I think it would call out that we should probably have the `SmartOnly` field in this struct as well.

Reply:
"smart" stuff removed in [f8ccbd0](https://github.com/hashicorp/nomad/pull/20130/commits/f8ccbd02da0141a2c3f3d18ae6b0635bd730178d)

but now there's an `include_children` option to which this comment still applies.

so I should either make this more complete and proper, or alternatively, exclude it from the `api` package altogether?  perhaps that could be a follow-on...  as it stands, this new endpoint is primarily for the UI to consume.

Reply:
`IncludeChildren` is on this struct now, and split out `NamespacedID` for the "TODO: proper type" comment, in [5e641a5](https://github.com/hashicorp/nomad/pull/20130/commits/5e641a56c6ed5724d5ca06b23ffaa8311fe39d0b)